test/integration.test.js

```diff
@@ -0,0 +1,439 @@
+'use strict';
```

Comment:
Can we make a `test/integration` directory so we can put this file in `test/integration/tests.js` so we can break up the tests into multiple files in the future